src/useEffectEvent.ts

```diff
@@ -1,5 +1,9 @@
 import {useCallback, useInsertionEffect, useRef} from 'react'
 
+function forbiddenInRender() {
+  throw new Error("A function wrapped in useEffectEvent can't be called during rendering.")
+}
```

Comment:
This is to replicate the behavior [on mount](https://github.com/facebook/react/blob/cd90a4d8c0d5dbaa8ab61e839b112b1518d5058f/packages/react-reconciler/src/ReactFiberHooks.js#L2867-L2871), and [on update](https://github.com/facebook/react/blob/cd90a4d8c0d5dbaa8ab61e839b112b1518d5058f/packages/react-reconciler/src/ReactFiberHooks.js#L2884-L2888)